### PR TITLE
Performance improvements (lazy loading) for booking pages

### LINKED ISSUE #1664: improve lighthouse score (google page speed) of mobile pages
https://pagespeed.web.dev/report?url=https%3A%2F%2Fcal.com%2Fpeer

[cal.com/peer](https://cal.com/peer) has a great lighthouse score of 98 on desktop

image

but not on mobile:
image

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?
1. It improves the Mobile PSI Score for Booking Page by more than twice. Should load really fast for low-end mobile users on slow connections.
2. It also fixes the issue of default gravatar not showing on booking page.

Note: Performance of `[user] `page is improved i.e. https://cal.com/hariom.` [user]/[type]` page would not improve
If this change works fine. Similar changes can be done on other pages.

The actual PSI Score should be better than this as HTTP2 isn't enabled in the deployment I am using for testing.
Screenshot 2022-02-09 at 6 17 12 PM

Fixes #1664

## Type of change

- [x] Performance Improvement (non-breaking change)
- [x] Bug Fix(non-breaking change)

## How should this be tested?
- [ ] Verify that the user page is loading fine and the event is bookable.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code and corrected any misspellings
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by PeerRich** [comment]
testing now! thank you for lazy loading some imports. @zomars was thinking about something similar

**Comment by PeerRich** [comment]
whats missing to go from draft to live? @hariombalhara

**Comment by hariombalhara** [comment]
Just self review is pending. If you have reviewed it. It's good to go from my side too.

File: package.json [hunk 1]
```diff
@@ -27,7 +27,9 @@
     "husky": "^7.0.1",
     "lint-staged": "^11.1.2",
     "prettier": "^2.5.1",
-    "prettier-plugin-tailwindcss": "^0.1.6",
+    "prettier-plugin-tailwindcss": "^0.1.6"
+  },
+  "dependencies": {
```
Comment (hariombalhara) [inline package.json:32]: It was required for my deployment to work on Heroku.
Heroku was throwing error that turbo is missing during start command

File: apps/web/components/ui/AvatarSSR.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+export type AvatarProps = {
+  user: Pick<User, "name" | "username" | "avatar"> & { emailMd5?: string };
+  className?: string;
+  size?: number;
+  title?: string;
+  alt: string;
+};
+
+// An SSR Supported version of Avatar component.
+// FIXME: title support is missing
```
Comment (hariombalhara) [inline apps/web/components/ui/AvatarSSR.tsx:15]: title isn't being used by [user] page.

**Review by hariombalhara — COMMENTED** [review]
It's ready to go live from my side @PeerRich

**Comment by PeerRich** [comment]
we moved to turbo repo which caused this branch to fail @hariombalhara. can you take a look at the new file structure? apps/web

**Comment by hariombalhara** [comment]
> 

I guess that issue is sorted now. All tests are passing.

**Review by emrysal — APPROVED** [review]

**Comment by PeerRich** [comment]
> > 
> 
> 
> 
> I guess that issue is sorted now. All tests are passing.

thank you! merged 👏

**Comment by hariombalhara** [comment]
Thanks. Much better but a lot of scope still.
@

**Pull request merged**